x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/ValueExtractor.java

```diff
@@ -26,12 +27,17 @@ interface ValueExtractor {
     void writeValue(BreakingBytesRefBuilder values, int position);
 
     static ValueExtractor extractorFor(ElementType elementType, TopNEncoder encoder, boolean inKey, Block block) {
-        if (false == (elementType == block.elementType() || ElementType.NULL == block.elementType())) {
+        if (false == (elementType == block.elementType()
+            || ElementType.NULL == block.elementType()
+            || ElementType.COMPOSITE == block.elementType())) {
             // While this maybe should be an IllegalArgumentException, it's important to throw an exception that causes a 500 response.
             // If we reach here, that's a bug. Arguably, the operators are in an illegal state because the layout doesn't match the
             // actual pages.
             throw new IllegalStateException("Expected [" + elementType + "] but was [" + block.elementType() + "]");
         }
+        if (elementType == ElementType.AGGREGATE_METRIC_DOUBLE && block.elementType() == ElementType.COMPOSITE) {
```

Comment:
With the changes to Block.read/write, we should always have AGGREGATE_METRIC_DOUBLE here.